hazelcast-jet-core/src/test/java/com/hazelcast/jet/core/metrics/MetricsTest.java

```diff
@@ -205,6 +210,193 @@ public void usingContextAsync() {
         job.join();
     }
 
+    @Test
+    public void availableDuringJobExecution() {
+        int generatedItems = 1000;
+
+        pipeline.drawFrom(TestSources.itemStream(1_000))
+                .withIngestionTimestamps()
+                .filter(l -> l.sequence() < generatedItems)
+                .map(t -> {
+                    Metrics.metric("total").increment();
+                    return t;
+                })
+                .drainTo(Sinks.logger());
+
+        Job job = instance.newJob(pipeline, JOB_CONFIG_WITH_METRICS);
+        assertTrueEventually(() -> {
+            assertMetricsProduced(job, "total", generatedItems);
+        });
+    }
+
+    @Test
+    public void when_metricsDisabled_then_unavailableDuringJobExecution() {
+        int generatedItems = 1000;
+
+        pipeline.drawFrom(TestSources.itemStream(1_000))
```

Comment:
Same undeterminism as with `availableDuringJobExecution`. Also you might want to use `assertTrueAllTheTime` to check that there aren't any metrics, ever.

Reply:
Ok, I will use `assertTrueAllTheTime` for 3 seconds instead of current `assertTrue`.